osu.Game/Screens/Play/DimmableStoryboard.cs

```diff
@@ -57,6 +57,14 @@ private void initializeStoryboard(bool async)
             if (!ShowStoryboard.Value && !IgnoreUserSettings.Value)
                 return;
 
+            double? storyboardLatestEventTime = storyboard.LatestEventTime;
+
+            if (!storyboard.ReplacesBackground && storyboardLatestEventTime != null)
```

Comment:
I have no idea what is going on here. Doesn't this mean you're drawing the background twice (something we are looking to avoid)?

At very least, this needs inline commenting explaining what the intention here is, because reading this code is not self-explanatory.

Reply:
This code updates the storyboard before the next line `drawableStoryboard = storyboard.CreateDrawable();` to create a `DrawableStoryboard` before loading every elements. It doesn't load or draw the background yet. I'll try to put an explaination.

Reply:
I understand what it's doing, but my question doesn't change. This change is incorrectly drawing a background layer a second time. In fact, it also hides any video layer as a result. You can test using https://osu.ppy.sh/beatmapsets/29664#osu/98345 of many other beatmaps.

If you are going to be drawing a background layer in the storyboard, then a few things need to happen:

- Background dim needs to be applied.
- The actual background (at the `Player` level) needs to be hidden - aka exactly what setting `ReplacesBackground` does here.